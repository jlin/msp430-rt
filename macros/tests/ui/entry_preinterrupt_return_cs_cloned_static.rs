#![no_main]

use msp430::interrupt::CriticalSection;
use msp430_rt_macros::entry;

fn init(cs: &CriticalSection<'static>) -> CriticalSection<'static> {
    cs.clone()
}

#[entry(interrupt_enable(pre_interrupt = init))]
fn main(_cs: CriticalSection) -> ! {
    loop {}
}
