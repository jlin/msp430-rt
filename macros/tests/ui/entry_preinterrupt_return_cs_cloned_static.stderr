error[E0597]: value does not live long enough
  --> tests/ui/entry_preinterrupt_return_cs_cloned_static.rs:10:1
   |
10 | #[entry(interrupt_enable(pre_interrupt = init))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
   | |                                              |
   | |                                              value dropped here while still borrowed
   | borrowed value does not live long enough
   | argument requires that `cs` is borrowed for `'static`
   |
   = note: this error originates in the attribute macro `entry` (in Nightly builds, run with -Z macro-backtrace for more info)
